### PR TITLE
fix calculation of memoryoffset shape 

### PR DESCRIPTION
**Pull request by sadolini**
### Details:
 Fixed MemoryOffset partial infer for 2 cases:
- after normalization layer - shape can be read from model with InputDim tag - fixed loader
- after affinecomponent - shape calculated in extractor and saved in out-size attribute - read it in split transformation and move to element_size for future use in partial infer

### Tickets:
 29401

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR **Not applicable**
* [x]  Transformation preserves original framework node names **Not applicable**

Validation:
* [x]  Unit tests **Not applicable**
* [x]  Framework operation tests **Not applicable**
* [x]  Transformation tests **Not applicable**
* [x]  Model Optimizer IR Reader check **Not applicable**

Documentation:
* [x]  Supported frameworks operations list **Not applicable**
* [x]  Guide on how to convert the **public** model **Not applicable**
* [x]  User guide update **Not applicable**

### CODE REVIEW

**Review by vgavrilo — APPROVED** [review]

File: model-optimizer/extensions/front/kaldi/split_recurrent_memoryoffset.py [hunk 1]
```diff
@@ -57,6 +57,13 @@ def find_and_replace_pattern(self, graph: Graph):
                 # MemoryOffset node is not in a recurrent block -- no splitting is needed
                 return
 
+            # check that node have information for future partial infer
```
Comment (rkazants) [inline model-optimizer/extensions/front/kaldi/split_recurrent_memoryoffset.py:60]: node have -> node has

File: model-optimizer/extensions/front/kaldi/split_recurrent_memoryoffset.py [hunk 2]
```diff
@@ -57,6 +57,13 @@ def find_and_replace_pattern(self, graph: Graph):
...
+            # element_size set in loader based on dimensions of previous layer that were read from Kaldi model
```
Comment (rkazants) [inline model-optimizer/extensions/front/kaldi/split_recurrent_memoryoffset.py:61]: element size is set?

File: model-optimizer/extensions/front/kaldi/split_recurrent_memoryoffset.py [hunk 3]
```diff
@@ -57,6 +57,13 @@ def find_and_replace_pattern(self, graph: Graph):
...
             if not offset_node.has_valid('element_size'):
-                raise Error("In a recurrent block 'element_size' for node {} is not set".format(offset_node.id))
+                # node have not set element_size but previous node can have information about its shape in out-size
+                # out-size set in extractor of some nodes like affinecomponent based on weights size
```
Comment (rkazants) [inline model-optimizer/extensions/front/kaldi/split_recurrent_memoryoffset.py:64]: also, please correct grammar

**Review by rkazants — APPROVED** [review]

**Review by pavel-esir — APPROVED** [review]
Looks good to me

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**